chia/farmer/farmer_api.py

```diff
@@ -565,30 +567,28 @@ async def request_signed_values(self, full_node_request: farmer_protocol.Request
             full_node_request.quality_string
         ]
 
-        foliage_block_data: Optional[SignatureRequestSourceData] = None
-        foliage_transaction_block_data: Optional[SignatureRequestSourceData] = None
-        include_source_data = False
+        message_data: Optional[List[Optional[SignatureRequestSourceData]]] = None
 
         if full_node_request.foliage_block_data is not None:
-            include_source_data = True
-            foliage_block_data = SignatureRequestSourceData(
-                uint8(SigningDataKind.FOLIAGE_BLOCK_DATA), bytes(full_node_request.foliage_block_data)
-            )
-
-        if full_node_request.foliage_transaction_block_data is not None:
-            assert foliage_block_data
-            include_source_data = True
-            foliage_transaction_block_data = SignatureRequestSourceData(
-                uint8(SigningDataKind.FOLIAGE_TRANSACTION_BLOCK),
-                bytes(full_node_request.foliage_transaction_block_data),
-            )
+            message_data = [
+                SignatureRequestSourceData(
+                    uint8(SigningDataKind.FOLIAGE_BLOCK_DATA), bytes(full_node_request.foliage_block_data)
+                ),
+                None
+                if full_node_request.foliage_transaction_block_data is None
```

Comment:
previously you, effectively, had an assert ensuring `full_node_request.foliage_transaction_block_data` is `not None` (line 579). if that condition truly holds, you could assert and simplify this a bit not scrapping the `None` case

Reply:
The assert was for `foliage_block_data` being `not None`, whenever we had a `full_node_request.foliage_transaction_block_data` present. Meaning `foliage_block_data` was always supposed to be present when `full_node_request.foliage_transaction_block_data` was. I simplified it by just doing it in the first branch when we check for `foliage_block_data`. Should be the same effect:

If `foliage_transaction_block_data` is present, then `foliage_block_data` must have also been present. In this case we already know it's present since we're doing the work in the same branch